build.py

```diff
@@ -549,15 +550,34 @@ def BuildAndInstall(context, buildArgs, stages):
                            "or choose a different location to install to."
                            .format(dir=dir))
                 sys.exit(1)
-        Print("""Success MayaUSD build and install !!!!""")
+        Print('Success MayaUsd build and install!')
 
-def RunTests(context,extraArgs):
-    RunCTest(context,extraArgs)
-    Print("""Success running MayaUSD tests !!!!""")
+def RunTests(context, extraArgs):
+    # Clean out any previous ctest logs before running ctest.
+    lastTestLog = os.path.join(context.buildDir, 'Testing', 'Temporary','LastTest.log')
+    if os.path.exists(lastTestLog):
+        os.remove(lastTestLog)
+    lastTestsFailedLog = os.path.join(context.buildDir, 'Testing', 'Temporary', 'LastTestsFailed.log')
+    if os.path.exists(lastTestsFailedLog):
+        os.remove(lastTestsFailedLog)
+
+    try:
+        RunCTest(context, extraArgs)
+    except RuntimeError as e:
+        PrintWarning('One (or more) tests failed, re-running only failed tests.\n')
+        try:
+            RunCTest(context, extraArgs, runOnlyFailed=True)
+        except:
+            PrintWarning('One (or more) tests failed on re-run!\n')
+            raise e # Re-raise the original failure.
+        else:
+            Print('Successfully re-ran failed tests!')
```

Comment:
The RunCTest() calls Run() which throws RuntimeError if an error was encountered executing the command. So catch that and re-run only the failed tests.